src/Mono.Android/Android.Runtime/OutputStreamInvoker.cs

```diff
@@ -9,25 +9,26 @@ public class OutputStreamInvoker : Stream
 
 		public OutputStreamInvoker (Java.IO.OutputStream stream)
 		{
+			if (stream == null)
+				throw new ArgumentNullException (nameof (stream));
+
 			this.BaseOutputStream = stream;
 		}
 
 		public override void Close ()
 		{
-			BaseOutputStream.Close ();
+			JavaCallWrapper.Call (() => BaseOutputStream.Close (), JavaCallWrapper.MakeIOException);
```

Comment:
Philosophical question is, do we really want to be creating all these delegates, or should we instead "manually inline" the try/catch blocks at each call site?

This is certainly *nicer to look at* -- though "nicer to *reason* about" is debatable -- but this "niceness" requires wrapping `BaseOutputStream.Close()` with a delegate instance (GC heap allocation!) and virtual method call (more  overhead).

How much overhead is that?  I don't know.  Presumably not *lots*, but it's not *zero* either.

Thus I *am* rather suggesting that we do:

```csharp
try {
    BaseOutputStream.Close ();
}
catch (Java.Lang.Throwable t) when (AndroidRuntime.BoundExceptionType == BoundExceptionType.System) {
    return JavaCallWrapper.MakeIOException (t.Message, t);
}
```

Reply:
I'm torn on this one. Thought about it too, and I chose the DRY approach since I prefer it this way, but yes, there's some overhead. OTOH, it's an exception so the overhead doesn't really matter that much, it's slow as molasses anyway. But maybe you're right and we should be more "explicit" in handling all of this...